modules/app/src/main/java/com/enonic/xp/app/contentstudio/widget/MediaRenderingBean.java

```
@@ -0,0 +1,433 @@
+                return new ResponseMapper( responseBuilder.cacheControl( cacheControl ).build() );
+            }
+        }
+
+        responseBuilder = resolveResponseFromContentType( content, size );
+        if ( responseBuilder != null )
+        {
+            return new ResponseMapper( responseBuilder.build() );
+        }
+        else
+        {
+            throw new WebApplicationException( Response.Status.NOT_FOUND );
+        }
+    }
+
+    private Response.ResponseBuilder resolveResponseFromContentType( final Content content, final int size )
+        throws IOException
+    {
+        final ContentType superContentTypeWithIcon = resolveSuperContentTypeWithIcon( content.getType() );
+        if ( superContentTypeWithIcon == null || superContentTypeWithIcon.getIcon() == null )
+        {
+            return null;
+        }
+        final Icon icon = superContentTypeWithIcon.getIcon();
+
+        final String mimeType = icon.getMimeType();
+        final byte[] contentImage = HELPER.readIconImage( icon, size );
+
+        return Response.ok( ByteSource.wrap( contentImage ), mimeType );
+    }
+
+    private ContentType resolveSuperContentTypeWithIcon( final ContentTypeName contentTypeName )
+    {
+        ContentType contentType = getContentType( contentTypeName );
+        while ( contentType != null && contentType.getIcon() == null )
+        {
+            contentType = getContentType( contentType.getSuperType() );
+        }
+        return contentType;
+    }
+
+    private ContentType getContentType( final ContentTypeName contentTypeName )
+    {
+        if ( contentTypeName == null )
+        {
+            return null;
+        }
+        return this.contentTypeServiceSupplier.get().getByName( new GetContentTypeParams().contentTypeName( contentTypeName ) );
+    }
+
+    private ScaleParams parseScaleParam( final Media media, final String scale, final int size )
+    {
+        if ( scale == null )
```

I would say this small checks are polluting the method, can be removed by editing the calling line:
`
final ScaleParams scaleParam = scale == null ? null : parseScaleParam( media, scale, sizeParam );`
The same applies for couple other places in this file. That also makes code coverage a little easier.

---

Ternary operators are usually harder to read than a good old fail fast 😉 

Also, it is used in 3 places right after that check, so it's better to check it in one place first:
```
        final int pos = scale.indexOf( ":" );
        final String horizontalProportion = scale.substring( 0, pos );
        final String verticalProportion = scale.substring( pos + 1 );
```